vitals/BatteryTemperatureValidator.java

```diff
@@ -0,0 +1,20 @@
+package vitals;
+
+public class BatteryTemperatureValidator implements IBatteryStateValidator{
+
+    private final float temperature;
+
+    public BatteryTemperatureValidator(float temperature) {
+        this.temperature = temperature;
+    }
+
+
+    @Override
+    public boolean validateBatteryState() {
+        if(this.temperature < 0 || this.temperature > 45) {
```

Comment:
SoC and temperature both have similar checks.
Any requirement on the range comparison will need to be implemented twice. Refactor to make changes in one place.